### PR TITLE
#166604435 User logout or redirect to login if not logged in

### PR DESCRIPTION
**Pull request by a-braham**
#### What does this PR do?
- Fix token decoding if not logged in
#### Description of Task to be completed?
- Check for missing token in localstorage
- Implement logout
- Implement a redirect to the homepage if logged out
#### How should this be manually tested?
- `$ git clone https://github.com/andela/ah-the-immortals-frontend`
- `$ git checkout ch-logout-166604435`
- `$ npm install`
- `$ npm run start:dev`
#### What are the relevant pivotal tracker stories?
[`#166604435`](https://www.pivotaltracker.com/n/projects/2326460/stories/166604435)

### CODE REVIEW

File: src/components/DummyPosts/NavBarDummy.jsx [hunk 1]
```diff
@@ -1,36 +1,64 @@
...
+
+class DammyNav extends Component {
+  static propTypes = {
+    logoutAction: PropTypes.func.isRequired,
+  };
+
+  state = {};
+
+  logout = (e) => {
+    e.preventDefault();
```
[the remark this answers is not in the document]
Comment (Ogutu-Brian) [reply src/components/DummyPosts/NavBarDummy.jsx]: Hello @a-braham ,
You can fix this by giving another key to the dispatch being mapped to props

**Review by Ogutu-Brian — APPROVED** [review]

File: src/components/DummyPosts/NavBarDummy.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (a-braham) [reply src/components/DummyPosts/NavBarDummy.jsx]: awesome

**Review by IssaIan — APPROVED** [review]
LGTM

**Review by matthenge — APPROVED** [review]
LGTM

**Pull request merged**